lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetColumnChunkIterator.java

```diff
@@ -80,7 +90,16 @@ public Page next()
         checkState(hasNext(), "No more data left to read in column (%s), metadata (%s), valueCount %s, dataPageCount %s", descriptor, metadata, valueCount, dataPageCount);
 
         try {
-            PageHeader pageHeader = readPageHeader();
+            byte[] pageHeaderAAD = null;
+            if (decryptionContext.isPresent()) {
+                if (dataPageCount == 0 && !dictionaryWasRead && metadata.hasDictionaryPage()) {
```

Comment:
`dataPageCount == 0` is not needed, there can be only one dictionary and its always first page if present.
So the rest of the condition is enough